### PR TITLE
Feature/extend azurerm serviceendpoint

### PR DESCRIPTION
**Pull request by a30004053**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

Issue Number: #527 

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Any relevant logs, error output, etc?

Ghostbin: https://pst.klgrth.io/paste/t67ed

### CODE REVIEW

**Comment by xuzhang3** [comment]
> [CLA assistant check](https://cla.opensource.microsoft.com/microsoft/terraform-provider-azuredevops?pullRequest=594) Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/terraform-provider-azuredevops?pullRequest=594) before we can accept your contribution.❌ a30004053 [sign now](https://cla.opensource.microsoft.com/microsoft/terraform-provider-azuredevops?pullRequest=594)You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/terraform-provider-azuredevops?pullRequest=594) it.

@a30004053  please sign the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/terraform-provider-azuredevops?pullRequest=594)

**Comment by a30004053** [comment]
@xuzhang3 - waiting on internal approval specifically re: this clause in the CLA
> Your Employer. References to “employer” in this Agreement include Your employer or anyone else for whom You are acting in making Your Submission, e.g. as a contractor, vendor, or agent. If Your Submission is made in the course of Your work for an employer or Your employer has intellectual property rights in Your Submission by contract or applicable law, You must secure permission from Your employer to make the Submission before signing this Agreement. In that case, the term “You” in this Agreement will refer to You and the employer collectively. If You change employers in the future and desire to Submit additional Submissions for the new employer, then You agree to sign a new Agreement and secure permission from the new employer before Submitting those Submissions.

**Comment by a30004053** [comment]
@xuzhang3,
I have secured the necessary approvals and have signed the CLA above.
Thank you for your patience!

**Comment by a30004053** [comment]
hello @xuzhang3, If everything looks good, is it okay to merge this PR? As per #527, this is a feature we would like to see in the mainline provider, we are currently running a forked provider in our organization.
Cheers!

File: website/docs/r/serviceendpoint_azurerm.html.markdown
Comment (xuzhang3) [inline website/docs/r/serviceendpoint_azurerm.html.markdown]:
```suggestion
### Manual AzureRM Service Endpoint(Subscription Scoped)
```

File: website/docs/r/serviceendpoint_azurerm.html.markdown
Comment (xuzhang3) [inline website/docs/r/serviceendpoint_azurerm.html.markdown]:
```suggestion
### Manual AzureRM Service Endpoint(ManagementGroup Scoped)
```

**Review by xuzhang3 — CHANGES_REQUESTED** [review]
@a30004053  A few changes requested.
When I run test  `TestAccServiceEndpointAzureRm_MgmtGrpCreateAndUpdate`, got error:
```log
=== CONT  TestAccServiceEndpointAzureRm_MgmtGrpCreateAndUpdate
    resource_serviceendpoint_azurerm_test.go:76: Step 1/2 error: Check failed: Check 5/9 error: azuredevops_serviceendpoint_azurerm.serviceendpointrm: Attribute 'azurerm_managment_group_id' expected to be set
--- FAIL: TestAccServiceEndpointAzureRm_MgmtGrpCreateAndUpdate (111.32s)

```

**Comment by a30004053** [comment]
@xuzhang3 - Apologies, I had overlooked the `t.Skip` in the acceptance tests

```bash
=== CONT  TestAccServiceEndpointAzureRm_MgmtGrpCreateAndUpdate
--- PASS: TestAccServiceEndpointAzureRm_MgmtGrpCreateAndUpdate (50.79s)
PASS
ok      github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/acceptancetests        50.902s
```

**Comment by xuzhang3** [comment]
```log
=== RUN   TestAccServiceEndpointAzureRm_CreateAndUpdate
=== PAUSE TestAccServiceEndpointAzureRm_CreateAndUpdate
=== RUN   TestAccServiceEndpointAzureRm_MgmtGrpCreateAndUpdate
=== PAUSE TestAccServiceEndpointAzureRm_MgmtGrpCreateAndUpdate
=== RUN   TestAccServiceEndpointAzureRm_AutomaticCreateAndUpdate
=== PAUSE TestAccServiceEndpointAzureRm_AutomaticCreateAndUpdate
=== CONT  TestAccServiceEndpointAzureRm_AutomaticCreateAndUpdate
=== CONT  TestAccServiceEndpointAzureRm_CreateAndUpdate
=== CONT  TestAccServiceEndpointAzureRm_MgmtGrpCreateAndUpdate
--- PASS: TestAccServiceEndpointAzureRm_MgmtGrpCreateAndUpdate (76.69s)
--- PASS: TestAccServiceEndpointAzureRm_AutomaticCreateAndUpdate (138.60s)
--- PASS: TestAccServiceEndpointAzureRm_CreateAndUpdate (213.23s)
PASS
ok      github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/acceptancetests        213.729s

```

**Review by xuzhang3 — APPROVED** [review]
@a30004053 LGTM

**Pull request merged**
